src/native/libs/System.Security.Cryptography.Native.Apple/pal_x509.c

```diff
@@ -151,7 +151,7 @@ int32_t AppleCryptoNative_X509GetRawData(SecCertificateRef cert, CFDataRef* ppDa
     }
 
     *ppDataOut = SecCertificateCopyData(cert);
-    *pOSStatus = *ppDataOut == NULL ? errSecParam : noErr;
+    *pOSStatus = *ppDataOut == NULL ? errSecParam : (OSStatus)noErr;
```

Comment:
This feels like a weird one to me, or the Clang compiler being a little confused. We use `noErr` all over the place but only complained in this place.

The reason it complains here is it is the only place we don't assign it to some OSStatus as an intermediate, and use it in an expression directly mixing it with an OSStatus. `noErr` is an anonymous enum in MacTypes.h.

The alternative is to use `errSecSuccess`. This is functionally the same as `noErr` (both have a value of 0), but I opted for the smaller change here. It would be a tad weird to use errSecSuccess in this function, but keep using noErr everywhere else, and changing everything seemed unnecessary.